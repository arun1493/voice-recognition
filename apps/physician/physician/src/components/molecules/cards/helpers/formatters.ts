import i18n from "i18next";
import _get from "lodash/get";
import { EMPTY_OBJECT } from "@suki-web/suki-business";

import {
  Allergy,
  AllergyReaction,
  CardMedicalHistoryProps,
  CardSurgicalHistoryProps,
  ParagraphCard,
  Problem
} from "../../../../types";
import { CardType } from "../../../../constants";

const formatParagraphCardData = (data: CardMedicalHistoryProps | CardSurgicalHistoryProps | any, cardType: string) => {
  const paragraphData: Array<ParagraphCard> = [];

  switch (cardType) {
    case CardType.ALLERGIES: {
      const { allergies, nkda } = data || EMPTY_OBJECT;

      allergies &&
        allergies.forEach((allergyItem: Allergy) => {
          const { allergen_name, reactions } = allergyItem || EMPTY_OBJECT;
          let allergyReactionContent = "";
          reactions &&
            reactions.forEach((allergyItem: AllergyReaction) => {
              const { reaction_name } = allergyItem;
              if (reaction_name) allergyReactionContent = allergyReactionContent.concat(reaction_name, ", ");
            });

          allergyReactionContent = allergyReactionContent.slice(0, -2); // to remove comma at the end of string

          if (allergen_name)
            paragraphData.push({
              title: allergen_name,
              content: allergyReactionContent
            });
        });

      if (nkda)
        paragraphData.push({
          content: i18n.t("pages.note.cards.errors.allergiesNkda")
        });
      break;
    }
    case CardType.PAST_MEDICAL_HISTORY: {
      const { active_problems } = data || EMPTY_OBJECT;
      active_problems &&
        active_problems.forEach((activeProblem: Problem) => {
          const description = _get(activeProblem, "note");
          const error = _get(activeProblem, "error");
          const title = _get(activeProblem, "diagnosis.icd_description");
          if (!error) paragraphData.push({ title, content: description });
        });
      break;
    }

    case CardType.PAST_SURGICAL_HISTORY: {
      data &&
        data.forEach((surgicalHistoryItem: Problem) => {
          const { description, note, error } = surgicalHistoryItem;
          if (!error) paragraphData.push({ title: description, content: note });
        });
      break;
    }

    default:
      break;
  }
  return paragraphData;
};

export { formatParagraphCardData };
