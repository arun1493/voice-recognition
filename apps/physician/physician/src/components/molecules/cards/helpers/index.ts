export * from "./labels";
export * from "./formatters";
