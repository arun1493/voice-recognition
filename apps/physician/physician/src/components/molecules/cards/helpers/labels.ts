import i18n from "i18next";

import { CardType, VITAL_SIGNS } from "../../../../constants";

const getCardTitle = (cardType: string) => {
  switch (cardType) {
    case CardType.ALLERGIES:
      return i18n.t("pages.note.cards.titles.allergies");
    case CardType.CURRENT_MEDICATION:
      return i18n.t("pages.note.cards.titles.medication");
    case CardType.PAST_MEDICAL_HISTORY:
    case CardType.PAST_MEDICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.titles.medicalHistory");
    case CardType.PAST_SURGICAL_HISTORY:
    case CardType.PAST_SURGICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.titles.surgicalHistory");
    case CardType.VITALS:
      return i18n.t("pages.note.cards.titles.vitals");
    default:
      return "";
  }
};

const getVitalSign = (sign: string) => {
  switch (sign) {
    case VITAL_SIGNS.BLOOD_PRESSURE:
      return i18n.t("pages.note.cards.vitalSigns.bloodPressure");
    case VITAL_SIGNS.BMI:
      return i18n.t("pages.note.cards.vitalSigns.bmi");
    case VITAL_SIGNS.HEIGHT:
      return i18n.t("pages.note.cards.vitalSigns.height");
    case VITAL_SIGNS.PULSE:
      return i18n.t("pages.note.cards.vitalSigns.pulse");
    case VITAL_SIGNS.WEIGHT:
      return i18n.t("pages.note.cards.vitalSigns.weight");
    case VITAL_SIGNS.HEART_RATE:
      return i18n.t("pages.note.cards.vitalSigns.heartRate");
    case VITAL_SIGNS.RESPIRATORY_RATE:
      return i18n.t("pages.note.cards.vitalSigns.respiratoryRate");
    case VITAL_SIGNS.TEMPERATURE:
      return i18n.t("pages.note.cards.vitalSigns.temperature");
    default:
      return null;
  }
};

const getErrorMessage = (cardType: string) => {
  switch (cardType) {
    case CardType.ALLERGIES:
      return i18n.t("pages.note.cards.errors.allergies");
    case CardType.CURRENT_MEDICATION:
      return i18n.t("pages.note.cards.errors.medication");
    case CardType.PAST_MEDICAL_HISTORY:
    case CardType.PAST_MEDICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.errors.medicalHistory");
    case CardType.PAST_SURGICAL_HISTORY:
    case CardType.PAST_SURGICAL_HISTORY_SHOW_ME:
      return i18n.t("pages.note.cards.errors.surgicalHistory");
    case CardType.VITALS:
      return i18n.t("pages.note.cards.errors.vitals");
    default:
      return "";
  }
};

export { getCardTitle, getErrorMessage, getVitalSign };
