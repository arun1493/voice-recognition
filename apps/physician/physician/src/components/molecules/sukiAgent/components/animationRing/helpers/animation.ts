import { RING_ANIMATION_DURATION, RING_BASE_OPACITY, RING_SCALE, SukiAgentState } from "../../../../../../constants";

function isValidSukiAgentStateForSingleRingAnimation(agentStatus: SukiAgentState, isAgentActive: boolean) {
  return (isAgentActive && agentStatus === SukiAgentState.AWAKE) || agentStatus === SukiAgentState.PROCESSING;
}

function isValidSukiAgentStateForMultipleRingAnimation(agentStatus: SukiAgentState, isAgentActive: boolean) {
  return (
    isAgentActive &&
    !isValidSukiAgentStateForSingleRingAnimation(agentStatus, isAgentActive) &&
    agentStatus === SukiAgentState.LISTENING
  );
}

function getDictationRingVariants(ringsCount: number) {
  const dictationRingVariants = [];
  for (let i = ringsCount; i > 0; i--) {
    const RING_VARIANTS = {
      oscillate: {
        opacity: RING_BASE_OPACITY - i * 0.2,
        scale: RING_SCALE + i * 0.01,
        transition: {
          yoyo: Infinity,
          duration: RING_ANIMATION_DURATION,
          type: "tween",
          delay: i * 0.1,
          ease: "easeInOut"
        }
      }
    };

    dictationRingVariants.push(RING_VARIANTS);
  }
  return dictationRingVariants;
}

export {
  getDictationRingVariants,
  isValidSukiAgentStateForMultipleRingAnimation,
  isValidSukiAgentStateForSingleRingAnimation
};
