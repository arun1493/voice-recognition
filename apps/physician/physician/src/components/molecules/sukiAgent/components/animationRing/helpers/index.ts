export * from "./ringSize";
export * from "./animation";
