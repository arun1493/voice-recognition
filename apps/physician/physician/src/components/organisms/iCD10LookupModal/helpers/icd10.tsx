import _isEmpty from "lodash/isEmpty";

const formatText = (text: string, highlight: string) => {
  if (_isEmpty(text)) return;
  const parts = text.split(new RegExp(`(${highlight})`, "gi"));
  return parts.map((part) => (part.toLowerCase() === highlight.toLowerCase() ? <strong>{part}</strong> : part));
};

const formatCode = (code: string) => {
  if (code?.length > 3) {
    return code?.substring(0, 3) + "." + code?.substring(3);
  }
  return code;
};

export { formatCode, formatText };
