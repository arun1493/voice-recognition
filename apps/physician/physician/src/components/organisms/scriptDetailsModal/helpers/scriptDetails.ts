import { ScreenSize } from "@suki-web/suki-ui";

const { SM, XS, XXS } = ScreenSize;

const getWidth = (breakPoint: ScreenSize) => {
  switch (breakPoint) {
    case XXS:
      return "100vw";
    case XS:
      return 484;
    case SM:
      return 664;
    default:
      return 782;
  }
};

const getHeight = (breakPoint: ScreenSize) =>
  breakPoint === XS || breakPoint === XXS ? "100vh" : "calc( 100vh - 56px )";

const getMaxHeight = (breakPoint: ScreenSize) => (breakPoint === XS || breakPoint === XXS ? "100vh" : 0);

export { getHeight, getMaxHeight, getWidth };
