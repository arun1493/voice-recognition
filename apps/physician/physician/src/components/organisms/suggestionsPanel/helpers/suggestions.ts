import i18n from "i18next";

import { SuggestionTypes } from "../../../../constants";

const getTextBySuggestion = (suggestion: string) => {
  switch (suggestion) {
    case SuggestionTypes.PL:
      return i18n.t("components.suggestionsPanel.pl");
    case SuggestionTypes.ED:
      return i18n.t("components.suggestionsPanel.ed");
    default:
      return "";
  }
};

export { getTextBySuggestion };
