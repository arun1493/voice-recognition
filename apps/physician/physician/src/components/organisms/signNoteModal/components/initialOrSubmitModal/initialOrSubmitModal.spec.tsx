import _noop from "lodash/noop";
import ReactDOM from "react-dom";
import { ReactPortal } from "react";
import "@testing-library/jest-dom/extend-expect";
import { cleanup } from "@testing-library/react";

import { InitialOrSubmitModal } from ".";
import { configureI18n } from "../../../../../helpers";
import { componentSetupWithStore } from "../../../../../hocs";
import { shouldMatchSnapshot } from "../../../../../helpers/snapshot.spec";

const setup = componentSetupWithStore((isInitial) => (
  <InitialOrSubmitModal
    closeModal={_noop}
    hasEnhancedReview={true}
    isAthena={false}
    isInitial={isInitial}
    saveToEnhancedReview={_noop}
    submitNote={_noop}
  />
));

describe("InitialOrSubmitModal initial step modal", () => {
  let baseElement: Element;
  let element: Element | null;

  beforeEach(() => {
    configureI18n();
    baseElement = setup(true).baseElement;
    element = baseElement.querySelector("[data-cy='initial-or-submit-modal']");
  });

  afterEach(() => {
    cleanup();
  });

  it("should render successfully", () => {
    expect(baseElement).toBeTruthy();
  });

  it("should contain correct title", () => {
    expect(element?.innerHTML).toContain("Sign note");
  });

  it("should contain correct text", () => {
    expect(element?.innerHTML).toContain("or");
    expect(element?.innerHTML).toContain("By e-signing, you accept Suki AI’s");
    expect(element?.innerHTML).toContain("terms.");
  });

  it("should contain correct buttons text", () => {
    expect(element?.innerHTML).toContain("Sign note");
    expect(element?.innerHTML).toContain("Enhanced review");
  });
});

describe("InitialOrSubmitModal submit step", () => {
  let baseElement: Element;
  let element: Element | null;

  beforeEach(() => {
    configureI18n();
    baseElement = setup(false).baseElement;
    element = baseElement.querySelector("[data-cy='initial-or-submit-modal']");
    ReactDOM.createPortal = jest.fn<ReactPortal, any>((element, node) => element);
  });

  afterEach(() => {
    cleanup();
  });

  it("should render successfully", () => {
    expect(baseElement).toBeTruthy();
  });

  it("should contain correct title", () => {
    expect(element?.innerHTML).toContain("Sign note");
  });

  it("should contain correct text", () => {
    expect(element?.innerHTML).toContain("By e-signing, you accept Suki AI’s");
    expect(element?.innerHTML).toContain("terms.");
  });

  it("should contain correct buttons text", () => {
    expect(element?.innerHTML).toContain("Confirm");
    expect(element?.innerHTML).not.toContain("Enhanced review");
  });

  it("should match snapshot", () => {
    shouldMatchSnapshot(InitialOrSubmitModal);
  });
});
