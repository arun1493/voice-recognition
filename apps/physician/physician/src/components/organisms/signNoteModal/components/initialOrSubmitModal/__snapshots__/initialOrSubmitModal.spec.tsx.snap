// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InitialOrSubmitModal submit step should match snapshot 1`] = `
Array [
  <div
    className="modalMaskstyled__StyledBackdrop-sc-169llx0-0 fRODrN"
    onClick={[Function]}
  />,
  <div
    className="modalstyled__StyledModal-sc-1u144rj-1 bKqWjs"
    data-cy="initial-or-submit-modal"
    height="auto"
    width={384}
  >
    <div
      aria-label="Close"
      className="modalstyled__StyledCloseButton-sc-1u144rj-0 bSHZet modal-close"
      height={56}
    >
      <div
        className="navigationIconstyled__StyledNavigationIcon-lwvkbw-0 imuqWK"
        data-cy=""
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <button
          className="iconstyled__StyledButton-ifk1iv-0 kgKJBV"
          onClick={[Function]}
        >
          <svg
            height={24}
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.70857 6.29289C7.31805 5.90237 6.68488 5.90237 6.29436 6.29289C5.90383 6.68342 5.90383 7.31658 6.29436 7.70711L10.5865 11.9993L6.2944 16.2914C5.90388 16.6819 5.90388 17.3151 6.2944 17.7056C6.68493 18.0961 7.31809 18.0961 7.70862 17.7056L12.0007 13.4135L16.2928 17.7056C16.6834 18.0961 17.3165 18.0961 17.7071 17.7056C18.0976 17.3151 18.0976 16.6819 17.7071 16.2914L13.4149 11.9993L17.7071 7.70711C18.0976 7.31658 18.0976 6.68342 17.7071 6.29289C17.3166 5.90237 16.6834 5.90237 16.2929 6.29289L12.0007 10.5851L7.70857 6.29289Z"
              fill="#999999"
              xmlns="http://www.w3.org/2000/svg"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      className="signNoteModalstyled__StyledModal-sc-1w1ailj-0 jwAZsQ"
    >
      <h5
        className="headerstyled__StyledH5-sc-1y4xvz5-4 iITAex signNoteModalstyled__StyledSignNoteHeader-sc-1w1ailj-1 huzdAo"
      >
        Sign note
      </h5>
      <button
        className="buttonstyled__StyledButton-ws70ng-0 dbAsMC initialOrSubmitModalstyled__StyledButton-sc-76mn2j-1 isevq"
        data-cy="button"
        disabled={false}
        type="button"
        width="320px"
      >
        <svg
          height={24}
          viewBox="0 0 32 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M31.8726 0.857266C31.7938 0.710284 31.6865 0.575539 31.5598 0.458568C31.5006 0.40392 31.4372 0.353152 31.3704 0.306827C31.33 0.278784 31.2884 0.252369 31.2457 0.227708C31.1834 0.191715 31.119 0.159458 31.053 0.131325C30.9192 0.0742612 30.7791 0.0341671 30.638 0.0142908C30.5857 0.00693362 30.5334 0.00234664 30.4811 0.000694137C30.3647 -0.00298548 30.249 0.00788795 30.1368 0.0351335C29.1703 0.269872 24.4109 6.94707 22.0856 10.313L26.1445 12.8116C28.1961 9.27719 32.2016 2.13589 31.9921 1.17723C31.9678 1.06596 31.927 0.958699 31.8726 0.857266ZM20.8415 12.1477C20.9203 12.0236 21.1147 11.7321 21.3962 11.3182L25.5296 13.8626C25.2754 14.2933 25.0947 14.5933 25.016 14.7174L22.2563 19.0683C21.8299 19.7404 21.3349 20.3673 20.7765 20.9434L17.0197 18.6308C17.3006 17.8838 17.6555 17.1707 18.0818 16.4985L20.8415 12.1477ZM16.3896 21.0449C16.4535 20.6317 16.5373 20.2255 16.6409 19.8272L19.8511 21.8033C19.533 22.0696 19.1992 22.3219 18.8503 22.5596L17.5414 23.4512L16.9491 23.8659C16.8117 24.0309 16.5266 24.0454 16.2896 23.8995C16.0526 23.7537 15.9409 23.4948 16.0312 23.3009L16.1499 22.5946L16.3896 21.0449ZM0.5 24H12.5667L13 22H0.5C0.223858 22 0 22.2239 0 22.5V23.5C0 23.7761 0.223857 24 0.5 24Z"
            fill="white"
            fillRule="evenodd"
          />
        </svg>
        <p
          className="textstyled__StyledText-sc-1vedkxi-0 jxLYkL"
          color="white"
          cursor="default"
          onClick={[Function]}
          size="xl"
        >
          Confirm
        </p>
      </button>
      <div
        className="termsCaptionstyled__Terms-z6umdk-0 ifttTJ"
      >
        By e-signing, you accept Suki AIâ€™s
        <u
          className="termsCaptionstyled__TermsButton-z6umdk-1 jeLHLN"
          onClick={[Function]}
        >
          terms
          .
        </u>
      </div>
    </div>
  </div>,
]
`;
